sru/release-27.14/test-migrate-user-config.sh

```diff
@@ -0,0 +1,206 @@
+    echo "-------------------------------------------"
+
+    lxc delete --force $name
+    echo "###########################################"
+}
+
+function test_apt_news_false_upgrade {
+    series=$1
+    old_version=$2
+    echo -e "\n\n###########################################"
+    echo "## $series: $old_version -> 27.13.4: ua_config changes preserved in new user-config"
+    echo "###########################################"
+    name=$(echo $series-$old_version | tr .~ -)
+
+    echo -e "\n-------------------------------------------"
+    echo "** launching container"
+    echo "-------------------------------------------"
+    lxc launch -q ubuntu-daily:$series $name
+    sleep 5
+    lxc exec $name -- apt-get update > /dev/null
+    echo "-------------------------------------------"
+
+    install_old_version $name $series $old_version
+
+    echo -e "\n-------------------------------------------"
+    echo "** pro config set apt_news=false"
+    echo "-------------------------------------------"
+    lxc exec $name -- pro config set apt_news=false
+    echo "-------------------------------------------"
+    echo -e "\n-------------------------------------------"
+    echo "** Show uaclient.conf"
+    echo "-------------------------------------------"
+    lxc exec $name -- cat /etc/ubuntu-advantage/uaclient.conf
+    echo "-------------------------------------------"
+
+    upgrade_to_proposed $name "NO CONFFILE PROMPT"
+
+    echo -e "\n-------------------------------------------"
+    echo "** Show uaclient.conf"
+    echo "-------------------------------------------"
+    lxc exec $name -- cat /etc/ubuntu-advantage/uaclient.conf
+    echo "-------------------------------------------"
+    echo -e "\n-------------------------------------------"
+    echo "** Show user config"
+    echo "-------------------------------------------"
+    lxc exec $name -- pro config show
+    echo "-------------------------------------------"
+
+    lxc delete --force $name
+    echo "###########################################"
+}
+
+function test_uaclient_conf_changes_upgrade {
+    series=$1
+    old_version=$2
+    echo -e "\n\n###########################################"
+    echo "## $series: $old_version -> 27.13.4: preserve uaclient.conf changes"
+    echo "###########################################"
+    name=$(echo $series-$old_version | tr .~ -)
+
+    echo -e "\n-------------------------------------------"
+    echo "** launching container"
+    echo "-------------------------------------------"
+    lxc launch -q ubuntu-daily:$series $name
+    sleep 5
+    lxc exec $name -- apt-get update > /dev/null
+    lxc exec $name -- apt-get install debsums -y > /dev/null
+    echo "-------------------------------------------"
+
+    install_old_version $name $series $old_version
+
+    echo -e "\n-------------------------------------------"
+    echo "** make changes to uaclient.conf root"
```

Comment:
Should we have more changes to `uaclient.conf` here ? For example, adding the `features` and `settings_overrides` here ?

Reply:
yep that's a good idea!